parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -713,16 +715,40 @@ private static final int compareTwoByteArrays(byte[] array1, int offset1, int le
     if (array1 == array2 && offset1 == offset2 && length1 == length2) return 0;
     int min_length = (length1 < length2) ? length1 : length2;
     for (int i = 0; i < min_length; i++) {
-      if (array1[i + offset1] < array2[i + offset2]) {
-        return 1;
+      int value1 = array1[i + offset1];
+      int value2 = array2[i + offset2];
+      if (value1 != value2) {
+        return value2 - value1;
       }
-      if (array1[i + offset1] > array2[i + offset2]) {
-        return -1;
+    }
+    // check remainder
+    if (length1 == length2) {
+      return 0;
+    } else {
+      return length2 - length1;
+    }
+  }
+
+  public static final int compareTwoBinaryUnsigned(Binary first, Binary second) {
```

Comment:
This naming was based on the above `compareTwoByteBuffers`, `compareByteBufferToByteArray`, etc. If we want to rename all of these things for consistency maybe that should go in a separate PR?